### PR TITLE
Update fhir apis

### PR DESCRIPTION
**Pull request by ThishaniLucas**
- Read patient source endpoint from env
- Update uscore patient accelerator to externalize source connection and read the source endpoint from env.

### CODE REVIEW

File: fhir-apis/uscore-patient-api-healthcare-accelerator/uscore501_patient_connect.bal [hunk 1]
```diff
@@ -31,63 +41,92 @@ public isolated class Uscore501PatientSourceConnect {
...
+                searchParams[key] = param.value;
+            }
+        }
+        //convert search parameters to query string
+        string queryString = "";
+        foreach var [key, value] in searchParams.entries() {
+            // check if value is an array
+            if (value is string[]) {
+                foreach var v in value {
+                    queryString = queryString + key + "=" + v + "&";
```
Comment (AnjanaSenanayake) [inline fhir-apis/uscore-patient-api-healthcare-accelerator/uscore501_patient_connect.bal:81]: Shouldn't we use backtick templates for string concatenations?
https://learn-ballerina.github.io/best_practices/string_concat.html

File: fhir-apis/uscore-patient-api-healthcare-accelerator/uscore501_patient_connect.bal [hunk 2]
```diff
@@ -31,63 +41,92 @@ public isolated class Uscore501PatientSourceConnect {
...
-            identifier: [
-            ],
-            implicitRules: "https://www.hl7.org/fhir",
-            language: "en-US",
-            gender: "unknown", 
-            name: []};
-        return example;
+        http:Response|http:ClientError res =  sourceEp->get(READ + id);
+        if (res is http:ClientError) {
+            r4:FHIRError fhirError = r4:createFHIRError("Error occured when calling the source system.", r4:CODE_SEVERITY_ERROR,r4:TRANSIENT_EXCEPTION);
```
Comment (AnjanaSenanayake) [inline fhir-apis/uscore-patient-api-healthcare-accelerator/uscore501_patient_connect.bal:46]: Code needs to be reformatted?

File: fhir-apis/uscore-patient-api-healthcare-accelerator/uscore501_patient_connect.bal [hunk 3]
```diff
@@ -31,63 +41,92 @@ public isolated class Uscore501PatientSourceConnect {
...
+            } else {
+                queryString = queryString + key + "=" + <string>value + "&";
+            }
+        }
+        //remove last & if query string is not empty
+        if (queryString != "") {
+            queryString = "?" + queryString.substring(0, queryString.length() - 1);
+        }
+
+        http:Response|http:ClientError res = sourceEp->get(SEARCH + queryString);
```
Comment (AnjanaSenanayake) [inline fhir-apis/uscore-patient-api-healthcare-accelerator/uscore501_patient_connect.bal:92]: Ballerina encourage to use resource access action if possible. https://ballerina.io/learn/by-example/http-client-send-request-receive-response/

i.e: by default it is `get` and hence verb get is not required.

File: fhir-apis/uscore-patient-api-healthcare-accelerator/uscore501_patient_connect.bal (hunk 1 above)
Comment (ThishaniLucas) [reply fhir-apis/uscore-patient-api-healthcare-accelerator/uscore501_patient_connect.bal:81]: All these suggestions should be added to the templates. The accelerators are created using them.

Comment (AnjanaSenanayake) [reply fhir-apis/uscore-patient-api-healthcare-accelerator/uscore501_patient_connect.bal:81]: ack

**Review by AnjanaSenanayake — APPROVED** [review]

**Review by nirmal070125 — APPROVED** [review]

**Pull request merged**